samples/remote_attestation/common/dispatcher.cpp

```diff
@@ -195,36 +362,66 @@ int ecall_dispatcher::generate_encrypted_message(uint8_t** data, size_t* size)
     }
 
     encrypted_data_size = sizeof(encrypted_data_buf);
-    if (m_crypto->Encrypt(
-            m_crypto->get_the_other_enclave_public_key(),
+    memset(encrypted_data_buf, 0x00, encrypted_data_size);
+    if (m_crypto->encrypt_gcm(
+            m_enclave_config->sym_key,
+            ++(m_enclave_config->sequence_number),
             m_enclave_config->enclave_secret_data,
             ENCLAVE_SECRET_DATA_SIZE,
+            iv_str,
             encrypted_data_buf,
-            &encrypted_data_size))
+            &encrypted_data_size,
+            tag_str))
     {
-        uint8_t* host_buf = (uint8_t*)oe_host_malloc(encrypted_data_size);
+        // Reason we allocate memory in the host is so that the encrypted data
+        // can be accessed by the hosts - aka real world scenario where we have
+        // 2 hosts and two enclaves. Allocate space for iv_str & tag_str to be
+        // appended after encrypted data
+        total_size = ENCLAVE_SECRET_DATA_SIZE * 2 + IV_SIZE;
```

Comment:
Not sure why this total_size is being calculated again.

Reply:
Good catch. I meant to remove the first calculation during declaration of total_sense.

Reply:
Done